src/utils/restApi/api.tsx

```diff
@@ -46,70 +49,79 @@ const getDesired = (data: DesiredType) => {
 };
 
 export const getContracts = async (
-  client: RestAPI,
+  client: RestClient,
   setData: Dispatch<SetStateAction<ITableData[] | null>>,
   setError: Dispatch<SetStateAction<string | null>>,
+  searchQuery: string,
 ) => {
   try {
     const range = contractsRange(`contractId;limit 10;offset 0;order desc`);
 
-    const allContracts = await client.getContracts({
-      tags: [`${env.NEXT_PUBLIC_DAPP_ID}`],
-      range,
-    });
-
-    const parsedContracts = contractSchema.safeParse(allContracts);
+    const tags =
+      searchQuery !== ""
+        ? [env.NEXT_PUBLIC_DAPP_ID + SWAP_TAG + searchQuery.toLocaleLowerCase()]
+        : [env.NEXT_PUBLIC_DAPP_ID];
 
-    if (!parsedContracts.success) {
-      throw new Error(parsedContracts.error.message);
-    }
+    const allContracts = await client.getContracts({ tags, range });
 
-    const validContracts = parsedContracts.data.headers.filter((contract) => {
-      if (
-        env.NEXT_PUBLIC_DAPP_ID in contract.tags &&
-        contract.tags[`${env.NEXT_PUBLIC_DAPP_ID}`]
-      ) {
-        const startDate =
-          contract.tags[`${env.NEXT_PUBLIC_DAPP_ID}`]?.startDate;
-        const expiryDate =
-          contract.tags[`${env.NEXT_PUBLIC_DAPP_ID}`]?.expiryDate;
-
-        return (
-          startDate &&
-          expiryDate &&
-          new Date(startDate) < new Date() &&
-          new Date(expiryDate) > new Date() &&
-          contract.status === "confirmed"
-        );
-      }
+    const succededContracts: ContractHeader[] = [];
+    allContracts.headers.forEach((header) => {
+      const parsedHeader = contractHeaderSchema.safeParse(header);
```

Comment:
what is this safeParse?

Reply:
@mikarasv had some problems with the interface of the contracts, we used [zod](https://zod.dev/?id=safeparse) to validate the data... `safeParse` doesn't throw errors if any contract fails

Reply:
what problem was encountered? in the request or in the response?

The response should be valid from the SDK, if they are not we have a bigger issue 😅. For requests, we have some dynamic guards (same as zod). Have you seen [this](https://input-output-hk.github.io/marlowe-ts-sdk/modules/_marlowe_io_language_core_v1.guards.html)? We may need to expose a similar `guards` module for `runtime-rest-client` and `runtime-lifecycle` for the complete request. In typescript this is only needed for the information that comes from outside the app (user input, reading a config file or similar). In JavaScript this is useful even for the code